From 75ce16cde57f8345a5e8dc73118f6a51211d5ff9 Mon Sep 17 00:00:00 2001
From: Daniel Borkmann <dborkman@redhat.com>
Date: Thu, 16 Jan 2014 15:19:48 -0500
Subject: [lib] crc32: reduce number of cases for crc32{, c}_combine

Message-id: <47acad715869dcc3782165a7d48244ec687ddd81.1389878943.git.dborkman@redhat.com>
Patchwork-id: 74353
O-Subject: [RHEL7 PATCH net 08/11] lib: crc32: reduce number of cases for crc32{, c}_combine
Bugzilla: 1054215
RH-Acked-by: Florian Westphal <fwestpha@redhat.com>
RH-Acked-by: Thomas Graf <tgraf@redhat.com>
RH-Acked-by: David S. Miller <davem@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1054215
Upstream Status: 165148396d8a220c0fb62e30101ea99b0223864e
Brew: http://brewweb.devel.redhat.com/brew/taskinfo?taskID=6885774
Tested: verified by QA in their IPVS lab and on beaker for basic tests

Upstream Commit:
commit 165148396d8a220c0fb62e30101ea99b0223864e
Author: Daniel Borkmann <dborkman@redhat.com>
Date:   Mon Nov 4 17:10:27 2013 +0100

    lib: crc32: reduce number of cases for crc32{, c}_combine

    We can safely reduce the number of test cases by a tenth.
    There is no particular need to run as many as we're running
    now for crc32{,c}_combine, that gives us still ~8000 tests
    we're doing if people run kernels with crc selftests enabled
    which is perfectly fine.

    Signed-off-by: Daniel Borkmann <dborkman@redhat.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Daniel Borkmann <dborkman@redhat.com>
Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/lib/crc32.c b/lib/crc32.c
index 3a1dfa8..70f00ca 100644
--- a/lib/crc32.c
+++ b/lib/crc32.c
@@ -1037,7 +1037,7 @@ static int __init crc32c_combine_test(void)
  int i, j;
  int errors = 0, runs = 0;
 
- for (i = 0; i < 100; i++) {
+ for (i = 0; i < 10; i++) {
   u32 crc_full;
 
   crc_full = __crc32c_le(test[i].crc, test_buf + test[i].start,
@@ -1131,7 +1131,7 @@ static int __init crc32_combine_test(void)
  int i, j;
  int errors = 0, runs = 0;
 
- for (i = 0; i < 100; i++) {
+ for (i = 0; i < 10; i++) {
   u32 crc_full;
 
   crc_full = crc32_le(test[i].crc, test_buf + test[i].start,
-- 
1.7.1